ZuluContent/Spells/Seventh/Polymorph.cs

```diff
@@ -73,164 +60,38 @@ public override bool CanCast()
-
-                            if (mt != null)
-                                mt.Rider = null;
-                        }
-
-                        Caster.BodyMod = m_NewBody;
-
-                        if (m_NewBody == 400 || m_NewBody == 401)
-                            Caster.HueMod = Race.DefaultRace.RandomSkinHue();
-                        else
-                            Caster.HueMod = 0;
-                        
-                        var val = (int) Caster.Skills[SkillName.Magery].Value;
-                        var duration = TimeSpan.FromSeconds(val > 120 ? 120 : val);
-
-                        BaseArmor.ValidateMobile(Caster);
-                        BaseClothing.ValidateMobile(Caster);
-
-                        Buff(Caster, m_NewBody, duration);
-                    }
-                }
-                else
-                {
-                    Caster.SendLocalizedMessage(1005559); // This spell is already in effect.
-                }
-            }
-
-            FinishSequence();
-        }
-
-        public static bool Buff(Mobile caster, int body, TimeSpan duration)
+        
+        public async Task CastAsync()
         {
-            if (caster.BeginAction(typeof(PolymorphSpell)))
-            {
-                if (body != 0)
-                {
-                    if (!((Body) body).IsHuman)
-                    {
-                        var mt = caster.Mount;
-
-                        if (mt != null)
-                            mt.Rider = null;
-                    }
-
-                    caster.BodyMod = body;
+            if (!Caster.CanBuff(Caster, icons: BuffIcon.Polymorph))
+                return;
 
-                    if (body == 400 || body == 401)
-                        caster.HueMod = Race.DefaultRace.RandomSkinHue();
-                    else
-                        caster.HueMod = 0;
+            var gump = new PolymorphGump(Caster, Categories);
+            var bodyId = await gump;
 
-                    BaseArmor.ValidateMobile(caster);
-                    BaseClothing.ValidateMobile(caster);
+            if (bodyId <= 0)
+                return;
 
-                    StopTimer(caster);
-                    
-                    Timer t = new InternalTimer(caster, duration);
-                    Timers[caster] = t;
+            var body = (Body) bodyId;
 
-                    t.Start();
-                    return true;
-                }
-            }
-            else
+            if (!body.IsHuman)
             {
-                caster.SendLocalizedMessage(1005559); // This spell is already in effect.
-            }
-            return false;
-        }
+                var mt = Caster.Mount;
 
-        public static bool StopTimer(Mobile m)
-        {
-            var t = (Timer) Timers[m];
-
-            if (t != null)
-            {
-                t.Stop();
-                Timers.Remove(m);
+                if (mt != null)
+                    mt.Rider = null;
```

Comment:
Curious, would this still let you have the mounted movement speed while morphed?

Reply:
That's removed now same with the BaseMount check for if you're polymorphed.

For reference that line is basically how to unmount someone.

Reply:
Mmm but as I remember in ZH u could still have the mounted movement speed while u were polymorphed.. was quite useful for PvP when u morphed into a bird and people could hardly see you :D

Reply:
haha yeah, I meant that the checks are removed, you can mount while polymorphed like you're talking about, it's the same as ZH3 now.

Reply:
ok sweet!